perses/tests/test_relative.py

```diff
@@ -711,10 +712,25 @@ def RepartitionedHybridTopologyFactory_energy_mutant(topology, chain, system, po
                                                use_14_nonbondeds=True)
 
     # Create geometry proposal
-    new_positions, logp_proposal = geometry_engine.propose(topology_proposal, positions, beta,
+    for i in range(5):
+        new_positions, logp_proposal = geometry_engine.propose(topology_proposal, positions, beta,
                                                                    validate_energy_bookkeeping=True)
-    logp_reverse = geometry_engine.logp_reverse(topology_proposal, new_positions, positions, beta,
+        logp_reverse = geometry_engine.logp_reverse(topology_proposal, new_positions, positions, beta,
                                                 validate_energy_bookkeeping=True)
+        # Check potential energy
+        platform = openmm.Platform.getPlatformByName('Reference')
+        integrator = LangevinIntegrator(temperature=temperature)
+        context = openmm.Context(topology_proposal.new_system, integrator, platform)
+        context.setPeriodicBoxVectors(*topology_proposal.new_system.getDefaultPeriodicBoxVectors())
+        context.setPositions(new_positions)
+        context.setVelocitiesToTemperature(temperature)
+        potential_energy = context.getState(getEnergy=True).getPotentialEnergy()
+
+        if potential_energy < 1e8 * unit.kilojoule_per_mole:
+            break
+        else:
+            if i == 4:
+                raise Exception("Was not able to generate a decent proposal after 5 iterations")
```

Comment:
This is nice! I believe a more readable way would be using a `for-else` statement here. I will be making the changes to accomplish that.

Reply:
Fixed in https://github.com/choderalab/perses/pull/828/commits/867b40e48d249213a83349d28b4123fe7c8637e1

Reply:
Cool, thanks! I didn't know about `for-else` statements!